src/Runner/Parallel/Process.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner\Parallel;
+
+use PhpCsFixer\Console\Command\FixCommand;
+use PhpCsFixer\Runner\RunnerConfig;
+use PhpCsFixer\ToolInfo;
+use React\ChildProcess\Process as ReactProcess;
+use React\EventLoop\LoopInterface;
+use React\EventLoop\TimerInterface;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\WritableStreamInterface;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Process\PhpExecutableFinder;
+
+/**
+ * Represents single process that is handled within parallel run.
+ * Inspired by:
+ *   - https://github.com/phpstan/phpstan-src/blob/9ce425bca5337039fb52c0acf96a20a2b8ace490/src/Parallel/Process.php
+ *   - https://github.com/phpstan/phpstan-src/blob/1477e752b4b5893f323b6d2c43591e68b3d85003/src/Process/ProcessHelper.php.
+ *
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ */
+final class Process
+{
+    // Properties required for process instantiation
+    private string $command;
+    private LoopInterface $loop;
+    private int $timeoutSeconds;
+
+    // Properties required for process execution
+    private ReactProcess $process;
+    private ?WritableStreamInterface $in = null;
+
+    /** @var false|resource */
+    private $stdErr;
+
+    /** @var false|resource */
+    private $stdOut;
+
+    /** @var callable(mixed[]): void */
+    private $onData;
+
+    /** @var callable(\Throwable): void */
+    private $onError;
+
+    private ?TimerInterface $timer = null;
+
+    public function __construct(string $command, LoopInterface $loop, int $timeoutSeconds)
```

Comment:
whenever I see both `__construct` and `static create()` - do we need both public? maybe only `create()` should public?

Reply:
I've made constructor private.